content/pandas/concepts/dataframe/terms/insert/insert.md

```diff
@@ -0,0 +1,132 @@
+|     `value`     | Scalar, Series, or array-like.                        | Content to be inserted as the new column.  Note that passing a single value will set that same value for all rows.                                                                                                                                                 |
+| `allow_duplicates` | bool. Default `False` | If `allow_duplicates` is set to `True`, the new column is allowed to have the same `column` value as a preexisting column. If `allow duplicates` is set to `False`, passing a column name that is associated with a preexisting column will throw a `ValueError`. |
+
+## Example
+
+In the following examples, the `.insert()` method is used with different parameters on the testResults DataFrame:
+
+```
+import pandas as pd
+ 
+# Create the DataFrame representing test results administered by Doric
+data = {'grade': [99, 76, 64, 83],
+        'name': ['Jarnathan', 'Sofina', 'Edgin Darvis', 'Simon the Sorcerer'],
+        'home': ['Frozenfar', 'Thay', 'Icewind Dale','Triboar']}
+ 
+testResults = pd.DataFrame(data)
+ 
+# Print the original DataFrame
+print("Original Test Results:")
+print(testResults)
+```
+
+Output:
+
+```
+Original Test Results:
+   grade                name          home
+0     99           Jarnathan     Frozenfar
+1     76              Sofina          Thay
+2     64        Edgin Darvis  Icewind Dale
+3     83  Simon the Sorcerer       Triboar
+```
+
+Doric forgot to insert the alignment of each person tested.  She uses `insert()` to add the alignment of each person as follows:
+
+```
+#insert new column
+testResults.insert(3, 'alignment', ['Lawful Good','Neutral Evil','Chaotic Good','Chaotic Good'])
+
+#print  the updated DataFrame
+print("Updated Test Results:")
+print(testResults)
+```
+
+Output:
+
+```
+Updated Test Results:
+   grade                name          home     alignment
+0     99           Jarnathan     Frozenfar   Lawful Good
+1     76              Sofina          Thay  Neutral Evil
+2     64        Edgin Darvis  Icewind Dale  Chaotic Good
+3     83  Simon the Sorcerer       Triboar  Chaotic Good
+```
+
+Szass Tam used *scrying* to learn Doric's password.  To confuse Doric, Szass decided to add an identical *grade* column as follows:
+
+```
+#insert the 'new' column
+testResults.insert(1, 'grade', [54,100,26,70],allow_duplicates=True)
+
+#print  the updated DataFrame
+print("Further Updated Test Results:")
+print(testResults)
+```
+
+Output:
+
+```
+   grade  grade                name          home     alignment
+0     99     54           Jarnathan     Frozenfar   Lawful Good
+1     76    100              Sofina          Thay  Neutral Evil
+2     64     26        Edgin Darvis  Icewind Dale  Chaotic Good
+3     83     70  Simon the Sorcerer       Triboar  Chaotic Good
+```
```

Comment:
Please add the printed statement "Further Updated Test Results:" at the start of the code.